### ISSUE TITLE
Regression: Search results omits HTML tags

### ISSUE THREAD

**Issue by SethFalco** [easyops-cn/docusaurus-search-local#455]
I'm looking at upgrading Docusaurus and the plugins we're using, but I think there's a regression between v0.40.1 and v0.41.0 which degrades the search experience.

It does not handle HTML tags in content well anymore.

Our documentation contains the following:

```md
Removes the [`<metadata>`](https://developer.mozilla.org/docs/Web/SVG/Element/metadata) element from the document.
```

## Before (v0.40.1)

This version kept the name of the HTML element, and omitted the Markdown syntax. The result was perfect in my opinion.

![](https://github.com/user-attachments/assets/5c1f8570-143f-483b-8bb4-fc5367777264)

## After (>= v0.41.0, including v0.44.5)

This version only keeps the backticks in the Markdown, and nothing inside it.

![](https://github.com/user-attachments/assets/ed20be06-ae89-4326-922e-9ffd8d5f033d)

**Comment by weareoutman** [comment]
<img width="447" alt="image" src="https://github.com/user-attachments/assets/bc392a42-8508-42b9-a66e-3a59ae49eb23">

I can't reproduce it.

According to your screenshot of after updating, the issue seems not to be related with the search plugin. It seems to be an issue with markdown renderer.

Under the hood, we scan the generated html files to build the search index, if the markdown renders correctly, the will be no backticks in the generated html (should be `<code></code>` instead).

**Comment by SethFalco** [comment]
TL;DR; I can confirm the problem is not from `@easyops-cn/docusaurus-search-local`. Thank you for investigating this for me.

---

I've done a git bisect in svg/svgo.dev, and found that this commit is where the bug started:
* https://github.com/svg/svgo.dev/pull/35/changes/aff017e10426557783ddd2e0e545cb97b8482f22

Testing further, when I only make the following change:

```diff
-     "@easyops-cn/docusaurus-search-local": "^0.40.1",
+     "@easyops-cn/docusaurus-search-local": "0.41.0",
```

It breaks! So the reason must be between:

* https://github.com/easyops-cn/docusaurus-search-local/compare/v0.40.1...v0.41.0

Then, I cloned `@easyops-cn/docusaurus-search-local`, installed the package via path, and did a git bisect here too—rebuilding my website between commit references. It reports the first broken commit to be:

* https://github.com/easyops-cn/docusaurus-search-local/commit/0c12b59866c6d9b7bfa342b4f87e40b01b5f7d48

So the regression is definietly introduces from the upgrade to `@easyops-cn/docusaurus-search-local`.

Then I manually inspected the HTML on that version, and indeed can see this isn't the fault of this library.

### MDX

```mdx
Removes the [`<metadata>`](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/metadata) element from the document.
```

### Built HTML

```html
<meta data-rh="true" name="description" content="Removes the `` element from the document.">
```

### Screenshot of Search Results

<img width="1352" height="165" alt="Image" src="https://github.com/user-attachments/assets/02ae7ace-df7a-431a-91d7-a5952ad8f8c5" />

---

While the problem was discovered from the plugin upgrade—it seems it's because the plugin incidentally/accidentally worked around the issue before, but no longer does. But the change revealed and underlying problem with my build pipeline, and is not a bug in the search plugin.

Docusaurus correctly renders the MDX in the page itself, but screws up the page metadata, which is the actual problem. And my metadata has been screwed since before this upgrade too! 😓 

---

Edit: Reported to [facebook/docusaurus](https://github.com/facebook/docusaurus/issues/11818).
